internal/broker/bind_create_test.go

```diff
@@ -101,6 +122,142 @@ func TestCreateBindingEndpoint(t *testing.T) {
+	})
+
+	t.Run("should return error when expiration_seconds is less than minExpirationSeconds", func(t *testing.T) {
+		// when
+		_, err := bindEndpoint.Bind(context.Background(), instanceID1, "binding-id-002", domain.BindDetails{
+			ServiceID:     "123",
+			PlanID:        fixture.PlanId,
+			RawParameters: json.RawMessage(`{"expirationSeconds": 100000}`),
+		}, false)
+		require.Error(t, err)
+
+		// then
+		assert.IsType(t, err, &apiresponses.FailureResponse{}, "Get request returned error of unexpected type")
+		apierr := err.(*apiresponses.FailureResponse)
+		assert.Equal(t, http.StatusBadRequest, apierr.ValidatedStatusCode(nil), "Get request status code not matching")
+	})
+}
+
+func TestCreateBindingExceedsAllowedNumberOfNonExpiredBindings(t *testing.T) {
+	// given
+	cfg := fixBindingConfig()
+	cfg.MaxBindingsCount = 1
+	bindEndpoint, db := prepareBindingEndpoint(t, cfg)
+	err := db.Bindings().Insert(&internal.Binding{
+		ID:                "existing-binding",
+		InstanceID:        instanceID1,
+		CreatedAt:         time.Now().Add(-2 * time.Hour),
+		UpdatedAt:         time.Time{},
+		ExpiresAt:         time.Now().Add(-1 * time.Hour),
+		Kubeconfig:        "abcd",
+		ExpirationSeconds: 600,
+		CreatedBy:         "",
+	})
+	require.NoError(t, err)
+
+	// when creating first binding
+	_, err = bindEndpoint.Bind(context.Background(), instanceID1, "binding-id-001", domain.BindDetails{
+		ServiceID:     "123",
+		PlanID:        fixture.PlanId,
+		RawParameters: json.RawMessage(`{}`),
+	}, false)
+
+	require.NoError(t, err)
+
+	// when creating second binding - expect error
+	_, err = bindEndpoint.Bind(context.Background(), instanceID1, "binding-id-002", domain.BindDetails{
+		ServiceID:     "123",
+		PlanID:        fixture.PlanId,
+		RawParameters: json.RawMessage(`{}`),
+	}, false)
+
+	assert.IsType(t, err, &apiresponses.FailureResponse{}, "Get request returned error of unexpected type")
+	apierr := err.(*apiresponses.FailureResponse)
+	assert.Equal(t, http.StatusBadRequest, apierr.ValidatedStatusCode(nil), "Get request status code not matching")
+
+}
+
+func TestCreateBindng(t *testing.T) {
+	bindEndpoint, _ := prepareBindingEndpoint(t, fixBindingConfig())
+
+	t.Run("should return error when expiration_seconds is less than minExpirationSeconds", func(t *testing.T) {
+		_, err := bindEndpoint.Bind(context.Background(), instanceID1, "binding-id-001", domain.BindDetails{
+			ServiceID:     "123",
+			PlanID:        fixture.PlanId,
+			RawParameters: json.RawMessage(`{"expiration_seconds": 10}`),
+		}, false)
+
+		// then
+		assert.IsType(t, err, &apiresponses.FailureResponse{}, "Get request returned error of unexpected type")
+		apierr := err.(*apiresponses.FailureResponse)
+		assert.Equal(t, http.StatusBadRequest, apierr.ValidatedStatusCode(nil), "Get request status code not matching")
+	})
+
+	t.Run("should return error when expiration_seconds is greater than maxExpirationSeconds", func(t *testing.T) {
```

Comment:
Should we test the case with the proper (not default) timeout?

Reply:
such test is in the envtest file, but I'll add it also here